src/Native/LdaNative/lda_engine.cpp

```diff
@@ -105,44 +101,30 @@ namespace lda {
             num_threads_ = std::max(1, (int)(uNumCPU - 2));
         }
         bAlphaSumMultiplied = false;
-        process_barrier_ = new SimpleBarrier(num_threads_);
-        atomic_stats_ = new LDAEngineAtomics();
-        data_block_ = new LDADataBlock(num_threads_);
-        model_block_ = new LDAModelBlock();
-        samplerQueue_ = new CBlockedIntQueue();
+        process_barrier_ = std::make_unique<SimpleBarrier>(num_threads_);
+        atomic_stats_ = std::make_unique<LDAEngineAtomics>();
+        data_block_ = std::make_unique<LDADataBlock>(num_threads_);
+        model_block_ = std::make_unique<LDAModelBlock>();
+        samplerQueue_ = std::make_unique<CBlockedIntQueue>();
 
         document_buffer_ = new int32_t*[num_threads_];
         for (int i = 0; i < num_threads_; i++)
             document_buffer_[i] = new int32_t[maxDocToken_ * 2 + 1];
 
-        likelihood_in_iter_ = nullptr;
         beta_sum_ = beta_ * V_;
     }
 
 
     LdaEngine::~LdaEngine()
     {
         //delete memory space
-        delete process_barrier_;
-        process_barrier_ = nullptr;
-
-        delete data_block_;
-        data_block_ = nullptr;
-
-        delete atomic_stats_;
-        atomic_stats_ = nullptr;
-
-        delete model_block_;
-        model_block_ = nullptr;
+        process_barrier_.reset(nullptr);
+        data_block_.reset(nullptr);
+        atomic_stats_.reset(nullptr);
+        model_block_.reset(nullptr);
+        samplerQueue_.reset(nullptr);
 
-        delete samplerQueue_;
-        samplerQueue_ = nullptr;
-
-        for (int i = 0; i < num_threads_; ++i)
-        {
-            delete samplers_[i];
-        }
-        delete[] samplers_;
```

Comment:
📝 This block was responsible for some of the crashes observed in CI, and especially the crashes prior to 0cbfa54b641663627b194775de30bac3ad35f803. The destructor is missing a null check for `samplers_`.